test/CliMenuTest.php

```diff
@@ -480,6 +480,66 @@ public function testAddCustomControlMapping() : void
         static::assertStringEqualsFile($this->getTestFile(), $this->output->fetch());
     }
 
+    public function testAddCustomControlMappingsThrowsExceptionWhenOverwritingExistingDefaultControls() : void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Cannot rebind this input');
+
+        $menu = new CliMenu('PHP School FTW', []);
+        $menu->addCustomControlMappings([
+            ' ' => function () {
+            }
+        ]);
+    }
+
+    public function testAddCustomControlMappingsThrowsExceptionWhenAttemptingToOverwriteAddedCustomControlMap() : void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Cannot rebind this input');
+
+        $menu = new CliMenu('PHP School FTW', []);
+        $menu->addCustomControlMappings([
```

Comment:
It took me a little while to figure this - you actually only send one element to the method, the second `c` key overwrites the first ! So it's actually not possible to call this method once and cause a rebind exception, you'd have to call it twice.

Reply:
Hahaha, of course ! Thank you !